client/src/Components/NewQuestion.js

```diff
@@ -21,48 +21,211 @@ export default function NewQuestion() {
+		setCd(e.target.value);
+		setQuestionToSet({ ...questionToSet,
+			code: `${e.target.value}`,
+
+		},
+		);
 	};
 	console.log( JSON.parse(JSON.stringify(Questions)));
 	const  answer_aChanged = (e) => {
 		setQuestionToSet({ ...questionToSet,
-			answers:{
+			answers:{ ...questionToSet.answers,
 				answer_a:`${e.target.value}`,
 			},
 		},
 		);
 	};
+	const  answer_bChanged = (e) => {
+		setQuestionToSet({ ...questionToSet,
+			answers:{ ...questionToSet.answers,
+				answer_b:`${e.target.value}`,
+			},
+		},
+		);
+	};
+	const  answer_cChanged = (e) => {
+		setQuestionToSet({ ...questionToSet,
+			answers:{ ...questionToSet.answers,
+				answer_c:`${e.target.value}`,
+			},
+		},
+		);
+	};
+	const  answer_dChanged = (e) => {
+		setQuestionToSet({ ...questionToSet,
+			answers:{ ...questionToSet.answers,
+				answer_d:`${e.target.value}`,
+			},
+		},
+		);
+	};
+	const  answer_eChanged = (e) => {
+		setQuestionToSet({ ...questionToSet,
+			answers:{ ...questionToSet.answers,
+				answer_e:`${e.target.value}`,
+			},
+		},
+		);
+	};
+
+	const  answer_fChanged = (e) => {
+		setQuestionToSet({ ...questionToSet,
+			answers:{ ...questionToSet.answers,
+				answer_f:`${e.target.value}`,
+			},
+		},
+		);
+	};
+	//checkboxes for correct answers
+	const answerASwitched=(event)=>{
+		setQuestionToSet({ ...questionToSet,
+			correct_answers:{ ...questionToSet.correct_answers,
+				answer_a_correct:event.target.checked?true:false,
+			},
+		});
+	};
+	const answerBSwitched=(event)=>{
+		setQuestionToSet({ ...questionToSet,
+			correct_answers:{ ...questionToSet.correct_answers,
+				answer_b_correct:event.target.checked?true:false,
+			},
+		});
+	};
+	const answerCSwitched=(event)=>{
+		setQuestionToSet({ ...questionToSet,
+			correct_answers:{ ...questionToSet.correct_answers,
+				answer_c_correct:event.target.checked?true:false,
+			},
+		});
+	};
+	const answerDSwitched=(event)=>{
+		setQuestionToSet({ ...questionToSet,
+			correct_answers:{ ...questionToSet.correct_answers,
+				answer_d_correct:event.target.checked?true:false,
+			},
+		});
+	};
+	const answerESwitched=(event)=>{
+		setQuestionToSet({ ...questionToSet,
+			correct_answers:{ ...questionToSet.correct_answers,
+				answer_e_correct:event.target.checked?true:false,
+			},
+		});
+	};
+	const answerFSwitched=(event)=>{
+		setQuestionToSet({ ...questionToSet,
+			correct_answers:{ ...questionToSet.correct_answers,
+				answer_f_correct:event.target.checked?true:false,
+			},
+		});
+	};
 	return (
 		<div className='col-12'>
 			<div>
-				<div>{questionToSet.answers.answer_a}</div>
+
+
+				<h1>code illustration</h1>
+				<textarea onKeyUp={(e) => codeChanged(e)} />
 				<h1>question,use markdown</h1>
 				<textarea onKeyUp={(e) => textChanged(e)} />
-
 				<h1>answers</h1>
 				<input onKeyUp={(e) => answer_aChanged(e)} placeholder="answer a" />
-				<input type='checkbox' />
+				<input type='checkbox' name={"answer a"} value={"a"} onChange={answerASwitched} />
```

Comment:
This will change later with new design decisions anyway, but for now - why are we doing both a text input and a checkbox for each answer?

Reply:
Checkbox is changing a property in question object indicating if this answer is correct or not and it is displayed in preview under mentors form inputs

Reply:
When input changes the text of the answer so text input has to deal with answers object, when input deals with correct_answer boolean object property: answer_a_correct:true